packages/builders/src/components/Component.ts

```diff
@@ -1,10 +1,15 @@
-import type { APIMessageComponent, ComponentType } from 'discord-api-types/v9';
+import type {
+	APIActionRowComponent,
+	APIMessageComponent,
+	APIModalComponent,
+	ComponentType,
+} from 'discord-api-types/v9';
 import type { JSONEncodable } from '../util/jsonEncodable';
 
 /**
  * Represents a discord component
  */
-export interface Component extends JSONEncodable<APIMessageComponent> {
+export interface Component extends JSONEncodable<APIMessageComponent | APIActionRowComponent<APIModalComponent>> {
```

Comment:
Why not use APIModalComponent
```suggestion
export interface Component extends JSONEncodable<APIMessageComponent | APIModalComponent> {
```

Reply:
I was messing with this and the only type that seems to make ts happy here is:
```
APIMessageComponent | APIActionRowComponent<APIModalComponent | APIMessageComponent> | APIModalComponent
```

I think this is because none of the dapi-types has `APIActionRowComponent<APIModalComponent | APIMessageComponent>` in them. Will probably make a -types PR.